packages/vite/src/migrations/update-16-5-0-change-ts-paths-plugin/change-ts-paths-plugin.ts

```diff
@@ -0,0 +1,65 @@
+import { Tree, getProjects, joinPathFragments } from '@nx/devkit';
+import { forEachExecutorOptions } from '@nx/devkit/src/generators/executor-options-utils';
+import { ViteBuildExecutorOptions } from '../../executors/build/schema';
+import { tsquery } from '@phenomnomnominal/tsquery';
+import { ImportDeclaration } from 'typescript';
+
+export default function update(tree: Tree) {
+  const projects = getProjects(tree);
+  forEachExecutorOptions<ViteBuildExecutorOptions>(
+    tree,
+    '@nx/vite:build',
+    (options, projectName) => {
+      const projectConfig = projects.get(projectName);
+      const config =
+        options.configFile || findViteConfig(tree, projectConfig.root);
+      if (!config || !tree.exists(config)) {
+        return;
+      }
+
+      const configContents = tree.read(config, 'utf-8');
+
+      const oldTsConfigPathPlugin =
+        tsquery.query<ImportDeclaration>(
+          configContents,
+          'ImportDeclaration:has(StringLiteral[value="vite-tsconfig-paths"])'
+        ) ?? [];
+
+      if (oldTsConfigPathPlugin.length === 0) {
+        return;
+      }
+
+      const importName =
+        oldTsConfigPathPlugin[0]?.importClause?.name?.text ??
+        'viteTsConfigPaths';
+      const updatedContent = tsquery.replace(
+        configContents,
+        `PropertyAssignment:has(Identifier[name="plugins"]) CallExpression:has(Identifier[name="${importName}"])`,
+        () => {
+          return `nxViteTsPaths()`;
+        }
+      );
+
+      const withImportChange = tsquery.replace(
+        updatedContent,
+        'ImportDeclaration:has(StringLiteral[value="vite-tsconfig-paths"])',
+        () => {
+          return "import { nxViteTsPaths } from '@nx/vite/plugins/nx-tsconfig-paths.plugin';";
+        }
+      );
+
+      // TODO(caleb): should we try to handle the commented out usage of viteTsConfigPaths()?
```

Comment:
Unsure if we need to migrate any commented out code?

Reply:
I think let's not, since it's more complicated and less straightforward, since we will not be able to use tsquery for that. So let's leave commented out code as is, is my opinion. Better left as is, rather than a mistake in migration.